### PR TITLE
subdir option for WebUI

### PR DESCRIPTION
**Pull request by nkottary**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Didn't this repo used to have CI? Doesn't seem like it ran on this PR.

**Comment by aviks** [comment]
> Didn't this repo used to have CI? Doesn't seem like it ran on this PR.

Looks like it had Travis CI, which probably doesnt work any more. Needs migration to GHA ci, I presume.

File: src/webui/gitutils.jl [hunk 1]
```diff
@@ -160,12 +161,18 @@ function gettreesha(
...
             run(`git -C $dest checkout $ref`)
-            match(r"tree (.*)", readchomp(`git -C $dest show --format=raw`))[1]
+
+            if isempty(subdir)
+                match(r"tree (.*)", readchomp(`git -C $dest show --format=raw`))[1], ""
+            elseif isdir(joinpath(dest, subdir))
+                match(r"tree (.*)\t", readchomp(`git -C $dest ls-tree -d HEAD $subdir`))[1], ""
+            else
+                nothing, "The sub-directory $subdir does not exist in this repository"
+            end
```
Comment (GunnarFarneback) [inline src/webui/gitutils.jl:171]: This looks fine but if desired all git operations after `git clone` could be simplified to reading the output of `git rev-parse $ref:$subdir`, which will return only the tree hash both when `subdir` is empty and non-empty.

File: src/webui/gitutils.jl [hunk 2]
```diff
@@ -162,10 +162,8 @@ function gettreesha(
             run(`git clone $url $dest`)
             run(`git -C $dest checkout $ref`)
 
-            if isempty(subdir)
-                match(r"tree (.*)", readchomp(`git -C $dest show --format=raw`))[1], ""
-            elseif isdir(joinpath(dest, subdir))
-                match(r"tree (.*)\t", readchomp(`git -C $dest ls-tree -d HEAD $subdir`))[1], ""
+            if isdir(joinpath(dest, subdir))
+                readchomp(`git -C $dest rev-parse HEAD:$subdir`)
```
Comment (GunnarFarneback) [inline src/webui/gitutils.jl]: This should still return an empty string as well?

**Comment by nkottary** [comment]
Tested registering this repo: https://github.com/registratortestorg/TestReg
Pull request for root package: https://github.com/JuliaComputing/TestRegistry/pull/10
Pull request for sub package: https://github.com/JuliaComputing/TestRegistry/pull/11

**Pull request merged**